miio/airqualitymonitor_miot.py

```diff
@@ -0,0 +1,254 @@
+import enum
+import logging
+
+import click
+
+from .click_common import command, format_output
+from .exceptions import DeviceException
+from .miot_device import MiotDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+MODEL_AIRQUALITYMONITOR_CGDN1 = "cgllc.airm.cgdn1"
+
+_MAPPING_CGDN1 = {
+    # Source https://miot-spec.org/miot-spec-v2/instance?type=urn:miot-spec-v2:device:air-monitor:0000A008:cgllc-cgdn1:1
+    # Environment
+    "humidity": {"siid": 3, "piid": 1},  # [0, 100] step 1
+    "pm25": {"siid": 3, "piid": 4},  # [0, 1000] step 1
+    "pm10": {"siid": 3, "piid": 5},  # [0, 1000] step 1
+    "temperature": {"siid": 3, "piid": 7},  # [-30, 100] step 0.00001
+    "co2": {"siid": 3, "piid": 8},  # [0, 9999] step 1
+    # Battery
+    "battery": {"siid": 4, "piid": 1},  # [0, 100] step 1
+    "charging_state": {
+        "siid": 4,
+        "piid": 2,
+    },  # 1 - Charging, 2 - Not charging, 3 - Not chargeable
+    "voltage": {"siid": 4, "piid": 3},  # [0, 65535] step 1
+    # Settings
+    "start_time": {"siid": 9, "piid": 2},  # [0, 2147483647] step 1
+    "end_time": {"siid": 9, "piid": 3},  # [0, 2147483647] step 1
+    "monitoring_frequency": {"siid": 9, "piid": 4},  # 1, 60, 300, 600, 0
+    "screen_off": {"siid": 9, "piid": 5},  # 15, 30, 60, 300, 0
+    "device_off": {"siid": 9, "piid": 6},  # 15, 30, 60, 0
+    "temperature_unit": {"siid": 9, "piid": 7},
+}
+
+
+class AirQualityMonitorMiotException(DeviceException):
+    pass
+
+
+class ChargingStateCGDN1(enum.Enum):
+    Unplugged = 0  # Not mentioned in the spec
+    Charging = 1
+    NotCharging = 2
+    NotChargable = 3
+
+
+class MonitoringFrequencyCGDN1(enum.Enum):
+    Every1Second = 1
+    Every1Minute = 60
+    Every5Minutes = 300
+    Every10Minutes = 600
+    NotSet = 0
```

Comment:
Are these the only values accepted by the device, or is it possible to use something like 120, too? If the latter, it'd be better to simply use an integer describing the frequency in seconds.

Reply:
It looks like the device accept:
`0...600 s` for monitoring frequency option
`0..300 s` for screen off option
`0..60 m` for device off option

But worth noting this can be considered as a kind of undocumented feature because the spec lists strictly possible options. So they could fix that one day on the firmware update. Another issue is the Xiaomi Home app shows wrongly what option is selected (see the attachment).

So it's up to you, whether to change it or not.

![photo_2021-01-08_14-41-48](https://user-images.githubusercontent.com/6884184/104021906-d63f2d00-51bf-11eb-8e06-ed6f6f5936d8.jpg)

Reply:
It's similar situation as with vacuum fan speeds, the official app may support only specific, hard-coded speeds but you can adjust them as you like. I'd say use integers instead of enums here, I prefer to give users that choice if they wish to use "unofficial" values (I do this all the time with my vacuum).

If you wish to do so, you could add methods/properties to return the "presets" based on the specified values, to help downstream users of the lib / document the "official" values.

Reply:
I made `int` to be accepted instead of enums. And similar to I believe the vacuum case I created properties with the official spec options. Let me please know is it now better.

Reply:
Fine for me. Please take a look at my comment how these are exposed though, that should be changed but this PR can be merged after that's done.